pdp/ast/jast/obligations.go

```diff
@@ -24,11 +25,43 @@ func (ctx context) unmarshalObligationItem(d *json.Decoder) (pdp.AttributeAssign
 			return newUnknownAttributeError(k)
 		}
 
-		e, err = ctx.unmarshalValueByType(a.GetType(), d)
+		delim, val, err := jparser.CheckObjectStartOrValue(d, "argument")
```

Comment:
As I recall, we've agreed on no backward compatibility. So obligation is always an expression. It makes no need of CheckObjectStartOrValue function.

Reply:
Yes, I know... I was very close to making it work when we discussed, so I dug deep. I thought it would be a good goal to avoid PAP changes.

My original plan was to have the numerical function changes (using expression only of obligations) as one PR, and the change to make it backward compatible a separate PR. This is so you can review to see whether the changes for backward compatibility is acceptable. (unfortunately, there was a lot conflicts when rebasing, so I bailed out...)

If you feel strongly that the backward compatibility changes should be pulled, I'll do that.